test/python/test_backend_name_resolution.py

```diff
@@ -54,44 +55,38 @@ def test_aliases(self, qe_token, qe_url):
                               backend_name=backend_name):
                 try:
                     backend_by_name = IBMQ.get_backend(backend_name)
-                except KeyError:
+                except QiskitBackendNotFoundError:
                     # The real name of the backend might not exist
                     pass
                 else:
                     backend_by_display_name = IBMQ.get_backend(display_name)
                     self.assertEqual(backend_by_name, backend_by_display_name)
                     self.assertEqual(backend_by_display_name.name(), backend_name)
 
-    def test_groups(self):
-        """Test that aggregate group names map to the first available backend
-        of their list of backends."""
-        aer_groups = Aer.grouped_backend_names()
-        for group_name, priority_list in aer_groups.items():
-            with self.subTest(group_name=group_name,
-                              priority_list=priority_list):
-                target_backend = _get_first_available_backend(priority_list)
-                if target_backend:
-                    self.assertEqual(Aer.get_backend(group_name),
-                                     Aer.get_backend(target_backend))
-
     def test_aliases_fail(self):
         """Test a failing backend lookup."""
-        self.assertRaises(LookupError, Aer.get_backend, 'bad_name')
+        self.assertRaises(QiskitBackendNotFoundError, Aer.get_backend, 'bad_name')
 
+    def test_aliases_return_empty_list(self):
+        """Test backends() return an empty list if name is unknown."""
+        self.assertEqual(Aer.backends("bad_name"), [])
```

Comment:
Ok, this is not a problem from your PR, but it raised up something that I would love to discuss with @diego-plan9 and @jaygambetta too. Are we fine returning an empty list if the backend doesn't exist? ... not sure if this could leads to bad patterns in error control from top-level code. What about throwing an exception in this case?

Reply:
Anyway, not blocking this PR because of this.

Reply:
According to the documentation `backends` returns a filtered list, so it makes sense for some criteria impossible to meet, that the list is empty:
https://github.com/Qiskit/qiskit-terra/blob/4d9be579405f607a0c779f558fac13bf0988c2b1/qiskit/backends/baseprovider.py#L61-L73

If you are getting only one backend, you can use `get_backend` which does precisely what you suggest:

https://github.com/Qiskit/qiskit-terra/blob/4d9be579405f607a0c779f558fac13bf0988c2b1/qiskit/backends/baseprovider.py#L39-L59

Reply:
I'm good with it being empty. @delapuente is spot on.